src/Console/Command/Migrate.php

```diff
@@ -37,8 +43,10 @@ protected function execute(InputInterface $input, OutputInterface $output)
         
         // environment
         $environment = $input->getOption('environment');
+        $configPath = $input->getOption('configuration');
+
         if (!$environment) {
```

Comment:
1) If environment passed you want to use default config in project root? 

2) Getter MUST be idempotent so it can't get arguments that mutate state of object. And

Reply:
My structure looks like:

```
bin/
app/
src/
...
```
Inside the the app folder I keep configs, migrations and so on. So
```
app/
-- mongodb/
---- migrations/
---- mongo-migrator.php
```

For that I need to pass for the binary custom path to the mongo-migrator.php. That's why I added new option.

Reply:
where did you place you composer.json. Good place for this config is like for any other be placed in project root.

and my questions was not about why you need this. I ask you why you implement it like that )))

Reply:
getConfig() may be called from any place in any command. Do you want to get different configs there?

Reply:
Yes, composer.json is placed in the project root. Like,
```
bin
app
src
tests
...
composer.json
package.json
webpack.config.js
... // and so on
```

The answer for question why I implement like that is too long.

Inside the app folder I keep any config, for example I have
```
app/
    postgres/
        phinx.php
        migrations/
    clickhouse/
        database.php
        migrations/
    yamls/
        ...
    ....
```

So, I wanted to keep the same structure with mongo db. Phinx has that ability, you can call binary from any folder and you can pass configuration file for migrations.

```
#!/bin/bash

cd "$( dirname "$( dirname "${BASH_SOURCE[0]}" )" )"

./vendor/bin/phinx --configuration=./app/db/phinx.php migrate $@
```

Reply:
@sokil yes